clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -314,93 +326,78 @@ public String toString() {
 
     // This is used to describe per-partition state in the MetadataResponse
     public static class PartitionMetadata {
-        private final Errors error;
-        private final int partition;
-        private final Node leader;
-        private final Optional<Integer> leaderEpoch;
-        private final List<Node> replicas;
-        private final List<Node> isr;
-        private final List<Node> offlineReplicas;
+        public final TopicPartition topicPartition;
```

Comment:
Why did we replace the partition id with the topic partition here?

Reply:
For simplicity I guess. It makes the `PartitionMetadata` object usable outside the context a `TopicMetadata`. It saved a little bit of conversion work here and there.